File: src/main/java/org/taktik/icure/services/external/rest/v1/dto/InsuranceDto.java
/*
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of iCureBackend.
 *
 * Foobar is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * (at your option) any later version.
 *
 * Foobar is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with iCureBackend.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.taktik.icure.services.external.rest.v1.dto;

import org.taktik.icure.services.external.rest.v1.dto.embed.AddressDto;

import java.util.Map;

public class InsuranceDto extends StoredDto {
    protected boolean privateInsurance = false;
    protected boolean hospitalisationInsurance = false;
    protected boolean ambulatoryInsurance = false;

    protected String code;
    protected String parent;
    protected AddressDto address= new AddressDto();

    protected java.util.Map<String, String> name;

    public boolean isPrivateInsurance() {
        return privateInsurance;
    }

    public void setPrivateInsurance(boolean privateInsurance) {
        this.privateInsurance = privateInsurance;
    }

    public boolean isHospitalisationInsurance() {
        return hospitalisationInsurance;
    }

    public void setHospitalisationInsurance(boolean hospitalisationInsurance) {
        this.hospitalisationInsurance = hospitalisationInsurance;
    }

    public boolean isAmbulatoryInsurance() {
        return ambulatoryInsurance;
    }

    public void setAmbulatoryInsurance(boolean ambulatoryInsurance) {
        this.ambulatoryInsurance = ambulatoryInsurance;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getParent() {
        return parent;
    }

    public void setParent(String parent) {
        this.parent = parent;
    }

    public AddressDto getAddress() {
        return address;
    }

    public void setAddress(AddressDto address) {
        this.address = address;
    }

    public Map<String, String> getName() {
        return name;
    }

    public void setName(Map<String, String> name) {
        this.name = name;
    }
}

File: src/main/kotlin/org/taktik/icure/services/external/rest/v1/dto/InsuranceDto.kt
/*
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of iCureBackend.
 *
 * iCureBackend is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2 as published by
 * the Free Software Foundation.
 *
 * iCureBackend is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with iCureBackend.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.taktik.icure.services.external.rest.v1.dto


import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.github.pozo.KotlinBuilder
import org.taktik.icure.services.external.rest.v1.dto.base.StoredDocumentDto
import org.taktik.icure.services.external.rest.v1.dto.embed.AddressDto

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
data class InsuranceDto(
        override val id: String,
        override val rev: String? = null,
        override val deletionDate: Long? = null,

        val name: Map<String, String> = mapOf(),
        val privateInsurance: Boolean = false,
        val hospitalisationInsurance: Boolean = false,
        val ambulatoryInsurance: Boolean = false,
        val code: String? = null,
        val agreementNumber: String? = null,
        val parent: String? = null, //ID of the parent
        val address: AddressDto = AddressDto()
) : StoredDocumentDto {
    override fun withIdRev(id: String?, rev: String) = if (id != null) this.copy(id = id, rev = rev) else this.copy(rev = rev)
    override fun withDeletionDate(deletionDate: Long?) = this.copy(deletionDate = deletionDate)
}
